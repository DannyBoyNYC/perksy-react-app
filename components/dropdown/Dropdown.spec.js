const React = require('react');
const { mount, shallow } = require('enzyme');

const Dropdown = require('./index').default;

describe('<Dropdown.* />', () => {
  let component;
  const requiredProps = {
    options: [
      {
        label: 'Foo',
        value: 'foo'
      },
      {
        label: 'Bar',
        value: 'bar'
      }
    ]
  };

  beforeEach(() => {
    jest.mock('react-select', () => 'Select');
  });

  describe('<Dropdown.SingleSelect />', () => {
    beforeEach(() => {
      component = shallow(<Dropdown.SingleSelect {...requiredProps} />).dive();
    });

    it('should render correctly with all required props', () => {
      expect(component).toMatchSnapshot();
    });

    it('should have the correct BEM class names', () => {
      expect(component).toHaveClassName('dropdown dropdown--single-select');
    });

    it('will set the default value of the select based on option value', () => {
      component = mount(
        <Dropdown.SingleSelect {...requiredProps} defaultValue="bar" />
      );
      expect(component.find('.dropdown__single-value')).toHaveText('Bar');
    });
  });

  describe('<Dropdown.MultiSelect />', () => {
    beforeEach(() => {
      component = shallow(<Dropdown.MultiSelect {...requiredProps} />).dive();
    });

    it('should render correctly with all required props', () => {
      expect(component).toMatchSnapshot();
    });

    it('should have the correct BEM class names', () => {
      expect(component).toHaveClassName('dropdown dropdown--multi-select');
    });
  });
});
