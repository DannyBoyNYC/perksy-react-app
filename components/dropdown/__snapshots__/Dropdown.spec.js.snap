// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown.* /> <Dropdown.MultiSelect /> should render correctly with all required props 1`] = `
<div
  className="dropdown dropdown--multi-select"
>
  <StateManager
    className="dropdown__select"
    classNamePrefix="dropdown"
    defaultInputValue=""
    defaultMenuIsOpen={false}
    defaultValue={Array []}
    isMulti={true}
    isSearchable={true}
    options={
      Array [
        Object {
          "label": "Foo",
          "value": "foo",
        },
        Object {
          "label": "Bar",
          "value": "bar",
        },
      ]
    }
  />
</div>
`;

exports[`<Dropdown.* /> <Dropdown.SingleSelect /> should render correctly with all required props 1`] = `
<div
  className="dropdown dropdown--single-select"
>
  <StateManager
    className="dropdown__select"
    classNamePrefix="dropdown"
    defaultInputValue=""
    defaultMenuIsOpen={false}
    defaultValue={Array []}
    isSearchable={true}
    options={
      Array [
        Object {
          "label": "Foo",
          "value": "foo",
        },
        Object {
          "label": "Bar",
          "value": "bar",
        },
      ]
    }
  />
</div>
`;
